lib/preload/ndb/preload.js

```diff
@@ -39,17 +38,22 @@ try {
 
   const sep = process.platform === 'win32' ? '\\' : '/';
   const stateFileName = `${nddStore}${sep}${process.pid}`;
-  fs.writeFileSync(stateFileName, JSON.stringify({
+  fs.writeFile(stateFileName, JSON.stringify({
     targetListUrl: targetListUrl,
     ppid: nddParentProcessId,
     data: nddData,
     argv: process.argv.concat(process.execArgv),
     cwd: process.cwd()
-  }));
-
-  inspector.close();
-  inspector.open(port, undefined, nddWaitForConnection !== '0');
-  delete process.breakAtStart;
+  }), err => {
```

Comment:
It is crucial to be synchronous here and prevent node from running any other code before we establish connection, otherwise we may miss some executions in debugger, so could you wrap existing synchronous call with try catch to check for errors?